src/app/core/registry/registry.service.ts

```diff
@@ -220,4 +258,224 @@ export class RegistryService {
+  }
+
+  public deselectMetadataField(field: MetadataField) {
+    this.store.dispatch(new MetadataRegistryDeselectFieldAction(field))
+  }
+
+  public deselectAllMetadataField() {
+    this.store.dispatch(new MetadataRegistryDeselectAllFieldAction())
+  }
+
+  public getSelectedMetadataFields(): Observable<MetadataField[]> {
+    return this.store.pipe(select(selectedMetadataFieldsSelector));
+  }
+
+  public createOrUpdateMetadataSchema(schema: MetadataSchema): Observable<MetadataSchema> {
+    const isUpdate = hasValue(schema.id);
+    const requestId = this.requestService.generateRequestId();
+    const endpoint$ = this.halService.getEndpoint(this.metadataSchemasPath).pipe(
+      isNotEmptyOperator(),
+      map((endpoint: string) => (isUpdate ? `${endpoint}/${schema.id}` : endpoint)),
+      distinctUntilChanged()
+    );
+
+    const serializedSchema = new DSpaceRESTv2Serializer(NormalizedObjectFactory.getConstructor(ResourceType.MetadataSchema)).serialize(schema as NormalizedMetadataSchema);
+
+    const request$ = endpoint$.pipe(
+      take(1),
+      map((endpoint: string) => {
+        if (isUpdate) {
+          const options: HttpOptions = Object.create({});
+          let headers = new HttpHeaders();
+          headers = headers.append('Content-Type', 'application/json');
+          options.headers = headers;
+          return new UpdateMetadataSchemaRequest(requestId, endpoint, JSON.stringify(serializedSchema), options);
+        } else {
+          return new CreateMetadataSchemaRequest(requestId, endpoint, JSON.stringify(serializedSchema));
+        }
+      })
+    );
+
+    // Execute the post/put request
+    request$.pipe(
+      configureRequest(this.requestService)
+    ).subscribe();
+
+    // Return created/updated schema
+    return this.requestService.getByUUID(requestId).pipe(
+      getResponseFromEntry(),
+      map((response: RestResponse) => {
+        if (!response.isSuccessful) {
+          if (hasValue((response as any).errorMessage)) {
+            this.notificationsService.error('Server Error:', (response as any).errorMessage, new NotificationOptions(-1));
+          }
+        } else {
+          this.showNotifications(true, isUpdate, false, { prefix: schema.prefix });
+          return response;
+        }
+      }),
+      isNotEmptyOperator(),
+      map((response: MetadataschemaSuccessResponse) => {
+        if (isNotEmpty(response.metadataschema)) {
+          return response.metadataschema;
+        }
+      })
+    );
+  }
+
+  public deleteMetadataSchema(id: number): Observable<RestResponse> {
+    return this.delete(this.metadataSchemasPath, id);
+  }
+
+  public clearMetadataSchemaRequests(): Observable<string> {
+    return this.halService.getEndpoint(this.metadataSchemasPath).pipe(
+      tap((href: string) => this.requestService.removeByHrefSubstring(href))
+    )
+  }
+
+  public createOrUpdateMetadataField(field: MetadataField): Observable<MetadataField> {
```

Comment:
Another large method without Typedocs.  (Technically all the new methods in this file should have Typedocs, but as most are one-liners, we could let them slide)